components/post_view/commented_on/commented_on.jsx

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import * as Utils from 'utils/utils.jsx';
+import {stripMarkdown} from 'utils/markdown';
+
+import CommentedOnFilesMessage from 'components/post_view/commented_on_files_message';
+
+export default class CommentedOn extends PureComponent {
+    static propTypes = {
+        displayName: PropTypes.string,
+        enablePostUsernameOverride: PropTypes.bool,
+        onCommentClick: PropTypes.func.isRequired,
+        post: PropTypes.object.isRequired,
+        actions: PropTypes.shape({
+            showSearchResults: PropTypes.func.isRequired,
+            updateSearchTerms: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    handleOnClick = () => {
+        const {actions, displayName} = this.props;
+
+        actions.updateSearchTerms(displayName);
+        actions.showSearchResults();
+    }
+
+    render() {
+        const {
+            displayName,
+            enablePostUsernameOverride,
+            post,
+        } = this.props;
+
+        let username = displayName;
+        if (
+            enablePostUsernameOverride &&
+            post.props &&
+            post.props.from_webhook &&
+            post.props.override_username
+        ) {
+            username = post.props.override_username;
+        }
+
+        let apostrophe = '\'s';
+        if (username.slice(-1) === 's') {
```

Comment:
Dropping the `s` for proper names is not grammatically correct based on the APA style guide (and other guides): http://blog.apastyle.org/apastyle/2013/06/forming-possessives-with-singular-names.html.

Reply:
So I guess we should find a way to make the dropping of "s" to be applied to username only.  In case of display names like fullname, nickname and first name, we'll keep as per your reference.
If it's fine with you, I'll create a separate ticket to address that.

Reply:
Usernames should not have the `s` dropped either since they're not nouns.

Reply:
Thanks @mkraft!

@jasonblais Before I update this PR, any thought on NOT dropping `s` on display name at "commented on".  I'm not sure if this has been discussed elsewhere and so just checking.

Note that this PR is just a refactor and without change to existing behavior.  Thanks!

![screen shot 2018-09-04 at 10 58 19 pm](https://user-images.githubusercontent.com/5334504/45039413-2b218480-b096-11e8-9688-9a8f1a16c4df.png)
`

Reply:
@esethna As briefly mentioned in our sprint planning, any thoughts on NOT dropping `s` after display name at "commented on". Thanks

Reply:
In the Chicago style, either way is correct, so we should decide which to follow everywhere. Regardless, this is an existing issue, so we should file a separate ticket for it instead of handling it here.

That does remind me however that this isn't properly localizable. For example, in French, it currently reads `le message de User's` with an extra `'s`. It gets even weirder for languages that don't use Latin character. I'll file a ticket for that

Reply:
Added as a note in the ticket, but we could rephrase the sentence with something like `Commented on a message from [username]`.

Or even simplier like `Replied to [username]`.

That way we don't need to worry about the `'s`.

Reply:
If we move the `'s` into the translation string and include it every time like Martin suggested, that would solve the problem. Regardless, we can do that separately since the code we're commenting on has existed for years.